tooling/debugger/src/context.rs

```diff
@@ -21,6 +21,128 @@ use thiserror::Error;
 use std::collections::BTreeMap;
 use std::collections::{hash_set::Iter, HashSet};
 
+#[derive(Clone, Debug, Eq, Hash, PartialEq, PartialOrd, Ord)]
+pub struct AddressMap {
+    // A Noir program is composed by
+    // `n` ACIR circuits
+    //       |_ `m` ACIR opcodes
+    //                |_ Acir call
+    //                |_ Acir Brillig function invocation
+    //                           |_ `p` Brillig opcodes
+    //
+    // We need to inline such structure to a one-dimension "virtual address" space
+    // We define the address space as a contiguous space where all ACIR and
+    // Brillig opcodes from all circuits are laid out
+    //
```

Comment:
I'd put this as part of the description of the whole struct, and expand on why we need this. Like

```
// The purpose of this structure is to map the opcode locations in ACIR circuits into
// a flat contiguous address space to be able to expose them to the DAP interface.
// In this address space, the ACIR circuits are laid out one after the other, and
// Brillig functions called from such circuits are expanded inline, replacing
// the `BrilligCall` ACIR opcode.
```